src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -572,18 +560,27 @@ impl ThinPool {
             self.resume()?;
         }
 
-        self.set_state(thin_pool_status);
+        self.set_state(self.thin_pool.status(get_dm())?);
 
+        Ok(should_save)
+    }
+
+    /// Check all filesystems on this thin pool and return which had their sizes
+    /// extended, if any. This method should not need to handle thin pool status
+    /// because it never alters the thin pool itself.
+    /// TODO: Put filesystem in read only if the MDV can't be saved after a pool
+    /// is extended.
+    pub fn check_fs(&mut self, pool_uuid: PoolUuid) -> StratisResult<ChangedProperties> {
+        let mut updated = ChangedProperties::default();
         for (name, uuid, fs) in self.filesystems.iter_mut() {
-            let save_mdv = fs.check()?;
-            if save_mdv {
-                if let Err(e) = self.mdv.save_fs(name, *uuid, fs) {
-                    error!("Could not save MDV for fs with UUID {} and name {} belonging to pool with UUID {}, reason: {:?}",
-                                uuid, name, pool_uuid, e);
-                }
+            let prop_diff = fs.check()?;
+            updated.add_fs_prop(*uuid, prop_diff);
+            if let Err(e) = self.mdv.save_fs(name, *uuid, fs) {
```

Comment:
Will this change make saving to the mdv occur too frequently? Previously the save was done only when something was judged to have changed, I think that should be preserved here.

Reply:
@mulkieran I agree, this was an oversight. I'll change this! I think the best solution may be passing back a boolean and an `Option` type that gets converted to a `StratFilesystemDiff::default()` when returned. What are your thoughts?

Reply:
@jbaublitz I may not be following you...```check()``` should return a pair of a bool, to indicate whether or not to update the metadata and an Option type, where ```None``` is converted to ```StratFilesystemDiff::default()``` but ```Some``` contains what?

Reply:
@mulkieran Let me play around with a couple more ideas and we can discuss it more tomorrow.